# **Stock Market Data Exploratory Data Analysis**

## **1. Data Collection and Cleaning**

**1.1. Extracting raw dataset**

In [1]:
from huggingface_hub import login

login(token="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")

In [3]:
from huggingface_hub import list_repo_files

files = list_repo_files(
    repo_id="ggaddam/OHLCV-1m",
    repo_type="dataset"
)

for f in files:
    print(f)

.gitattributes
README.md
data/ohlcv_1992-01.parquet
data/ohlcv_1992-02.parquet
data/ohlcv_1992-03.parquet
data/ohlcv_1992-04.parquet
data/ohlcv_1992-05.parquet
data/ohlcv_1992-06.parquet
data/ohlcv_1992-07.parquet
data/ohlcv_1992-08.parquet
data/ohlcv_1992-09.parquet
data/ohlcv_1992-10.parquet
data/ohlcv_1992-11.parquet
data/ohlcv_1992-12.parquet
data/ohlcv_1993-01.parquet
data/ohlcv_1993-02.parquet
data/ohlcv_1993-03.parquet
data/ohlcv_1993-04.parquet
data/ohlcv_1993-05.parquet
data/ohlcv_1993-06.parquet
data/ohlcv_1993-07.parquet
data/ohlcv_1993-08.parquet
data/ohlcv_1993-09.parquet
data/ohlcv_1993-10.parquet
data/ohlcv_1993-11.parquet
data/ohlcv_1993-12.parquet
data/ohlcv_1994-01.parquet
data/ohlcv_1994-02.parquet
data/ohlcv_1994-03.parquet
data/ohlcv_1994-04.parquet
data/ohlcv_1994-05.parquet
data/ohlcv_1994-06.parquet
data/ohlcv_1994-07.parquet
data/ohlcv_1994-08.parquet
data/ohlcv_1994-09.parquet
data/ohlcv_1994-10.parquet
data/ohlcv_1994-11.parquet
data/ohlcv_1994-12.parquet
dat

In [4]:
from huggingface_hub import list_repo_files

repo_id = "ggaddam/OHLCV-1m"

files = list_repo_files(
    repo_id=repo_id,
    repo_type="dataset"
)

parquet_files = [
    f for f in files
    if f.endswith(".parquet")
]

print(f"Number of parquet files: {len(parquet_files)}")

for f in parquet_files:
    print(f)

Number of parquet files: 403
data/ohlcv_1992-01.parquet
data/ohlcv_1992-02.parquet
data/ohlcv_1992-03.parquet
data/ohlcv_1992-04.parquet
data/ohlcv_1992-05.parquet
data/ohlcv_1992-06.parquet
data/ohlcv_1992-07.parquet
data/ohlcv_1992-08.parquet
data/ohlcv_1992-09.parquet
data/ohlcv_1992-10.parquet
data/ohlcv_1992-11.parquet
data/ohlcv_1992-12.parquet
data/ohlcv_1993-01.parquet
data/ohlcv_1993-02.parquet
data/ohlcv_1993-03.parquet
data/ohlcv_1993-04.parquet
data/ohlcv_1993-05.parquet
data/ohlcv_1993-06.parquet
data/ohlcv_1993-07.parquet
data/ohlcv_1993-08.parquet
data/ohlcv_1993-09.parquet
data/ohlcv_1993-10.parquet
data/ohlcv_1993-11.parquet
data/ohlcv_1993-12.parquet
data/ohlcv_1994-01.parquet
data/ohlcv_1994-02.parquet
data/ohlcv_1994-03.parquet
data/ohlcv_1994-04.parquet
data/ohlcv_1994-05.parquet
data/ohlcv_1994-06.parquet
data/ohlcv_1994-07.parquet
data/ohlcv_1994-08.parquet
data/ohlcv_1994-09.parquet
data/ohlcv_1994-10.parquet
data/ohlcv_1994-11.parquet
data/ohlcv_1994-12.parquet

In [5]:
import requests

base_url = f"https://huggingface.co/datasets/{repo_id}/resolve/main/"

file_sizes = {}

for f in parquet_files:
    url = base_url + f

    response = requests.head(
        url,
        allow_redirects=True
    )

    size = response.headers.get("Content-Length")

    if size:
        size_gb = int(size) / (1024**3)
        file_sizes[f] = size_gb
        print(f"{f}: {size_gb:.3f} GB")
    else:
        print(f"{f}: size unavailable")

data/ohlcv_1992-01.parquet: 0.031 GB
data/ohlcv_1992-02.parquet: 0.024 GB
data/ohlcv_1992-03.parquet: 0.024 GB
data/ohlcv_1992-04.parquet: 0.024 GB
data/ohlcv_1992-05.parquet: 0.020 GB
data/ohlcv_1992-06.parquet: 0.022 GB
data/ohlcv_1992-07.parquet: 0.023 GB
data/ohlcv_1992-08.parquet: 0.021 GB
data/ohlcv_1992-09.parquet: 0.022 GB
data/ohlcv_1992-10.parquet: 0.024 GB
data/ohlcv_1992-11.parquet: 0.025 GB
data/ohlcv_1992-12.parquet: 0.028 GB
data/ohlcv_1993-01.parquet: 0.029 GB
data/ohlcv_1993-02.parquet: 0.028 GB
data/ohlcv_1993-03.parquet: 0.030 GB
data/ohlcv_1993-04.parquet: 0.028 GB
data/ohlcv_1993-05.parquet: 0.027 GB
data/ohlcv_1993-06.parquet: 0.029 GB
data/ohlcv_1993-07.parquet: 0.028 GB
data/ohlcv_1993-08.parquet: 0.031 GB
data/ohlcv_1993-09.parquet: 0.030 GB
data/ohlcv_1993-10.parquet: 0.033 GB
data/ohlcv_1993-11.parquet: 0.032 GB
data/ohlcv_1993-12.parquet: 0.033 GB
data/ohlcv_1994-01.parquet: 0.035 GB
data/ohlcv_1994-02.parquet: 0.031 GB
data/ohlcv_1994-03.parquet: 0.037 GB
d

In [6]:
from huggingface_hub import list_repo_files

repo_id = "ggaddam/OHLCV-1m"

files = list_repo_files(
    repo_id=repo_id,
    repo_type="dataset"
)

parquet_files = sorted([
    f for f in files
    if f.startswith("data/ohlcv_") and f.endswith(".parquet")
], reverse=True)

#print("Latest:", parquet_files[0])
#print("Earliest:", parquet_files[-1])
#print("Number of months:", len(parquet_files))

Latest: data/ohlcv_2025-07.parquet
Earliest: data/ohlcv_1992-01.parquet
Number of months: 403


In [7]:
target_gb = 30
selected_files = []
total_gb = 0

for f in parquet_files:
  size_f = file_sizes[f]

  if total_gb + size_f <= target_gb:
    selected_files.append(f)
    total_gb += size_gb
  else:
    break

print(f"# Files: {len(selected_files)}")
print(f"Total GB: {total_gb:.2f} GB")

print("Latest:", selected_files[0])
print("Earliest:", selected_files[-1])

# Files: 83
Total GB: 30.13 GB
Latest: data/ohlcv_2025-07.parquet
Earliest: data/ohlcv_2018-09.parquet


**Observations:**

- The raw data range ranges between 2018-09 and 2025-07.

- Total size of the raw data is 30.13 GB

- Raw data is made up of 83 files.

**1.2. Downloading the working dataset**

In [9]:
target_gb = 13
selected_files = []
total_gb = 0

for f in parquet_files:
  size_f = file_sizes[f]

  if total_gb + size_f <= target_gb:
    selected_files.append(f)
    total_gb += size_gb
  else:
    break

print(f"# Files: {len(selected_files)}")
print(f"Total GB: {total_gb:.2f} GB")

print("Latest:", selected_files[0])
print("Earliest:", selected_files[-1])

# Files: 35
Total GB: 12.70 GB
Latest: data/ohlcv_2025-07.parquet
Earliest: data/ohlcv_2022-09.parquet


**Observations:**

- The working dataset size is 12.7 GB.

- The working dataset ranges from 09-20222 to 07-2025.

- The working contains 35 files.

In [14]:
from huggingface_hub import hf_hub_download

repo_id = "ggaddam/OHLCV-1m"


for i in range(len(selected_files)):
  file_path = hf_hub_download(
      repo_id=repo_id,
      filename=selected_files[i],
      repo_type="dataset"
  )

  print(file_path)

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2025-07.parquet


data/ohlcv_2025-06.parquet: reconstructing file:   0%|          |  0.00B /  354MB            

data/ohlcv_2025-06.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2025-06.parquet


data/ohlcv_2025-05.parquet: reconstructing file:   0%|          |  0.00B /  367MB            

data/ohlcv_2025-05.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2025-05.parquet


data/ohlcv_2025-04.parquet: reconstructing file:   0%|          |  0.00B /  405MB            

data/ohlcv_2025-04.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2025-04.parquet


data/ohlcv_2025-03.parquet: reconstructing file:   0%|          |  0.00B /  377MB            

data/ohlcv_2025-03.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2025-03.parquet


data/ohlcv_2025-02.parquet: reconstructing file:   0%|          |  0.00B /  336MB            

data/ohlcv_2025-02.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2025-02.parquet


data/ohlcv_2025-01.parquet: reconstructing file:   0%|          |  0.00B /  356MB            

data/ohlcv_2025-01.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2025-01.parquet


data/ohlcv_2024-12.parquet: reconstructing file:   0%|          |  0.00B /  356MB            

data/ohlcv_2024-12.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-12.parquet


data/ohlcv_2024-11.parquet: reconstructing file:   0%|          |  0.00B /  314MB            

data/ohlcv_2024-11.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-11.parquet


data/ohlcv_2024-10.parquet: reconstructing file:   0%|          |  0.00B /  355MB            

data/ohlcv_2024-10.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-10.parquet


data/ohlcv_2024-09.parquet: reconstructing file:   0%|          |  0.00B /  314MB            

data/ohlcv_2024-09.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-09.parquet


data/ohlcv_2024-08.parquet: reconstructing file:   0%|          |  0.00B /  345MB            

data/ohlcv_2024-08.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-08.parquet


data/ohlcv_2024-07.parquet: reconstructing file:   0%|          |  0.00B /  345MB            

data/ohlcv_2024-07.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-07.parquet


data/ohlcv_2024-06.parquet: reconstructing file:   0%|          |  0.00B /  290MB            

data/ohlcv_2024-06.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-06.parquet


data/ohlcv_2024-05.parquet: reconstructing file:   0%|          |  0.00B /  344MB            

data/ohlcv_2024-05.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-05.parquet


data/ohlcv_2024-04.parquet: reconstructing file:   0%|          |  0.00B /  353MB            

data/ohlcv_2024-04.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-04.parquet


data/ohlcv_2024-03.parquet: reconstructing file:   0%|          |  0.00B /  321MB            

data/ohlcv_2024-03.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-03.parquet


data/ohlcv_2024-02.parquet: reconstructing file:   0%|          |  0.00B /  330MB            

data/ohlcv_2024-02.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-02.parquet


data/ohlcv_2024-01.parquet: reconstructing file:   0%|          |  0.00B /  335MB            

data/ohlcv_2024-01.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2024-01.parquet


data/ohlcv_2023-12.parquet: reconstructing file:   0%|          |  0.00B /  325MB            

data/ohlcv_2023-12.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-12.parquet


data/ohlcv_2023-11.parquet: reconstructing file:   0%|          |  0.00B /  322MB            

data/ohlcv_2023-11.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-11.parquet


data/ohlcv_2023-10.parquet: reconstructing file:   0%|          |  0.00B /  339MB            

data/ohlcv_2023-10.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-10.parquet


data/ohlcv_2023-09.parquet: reconstructing file:   0%|          |  0.00B /  293MB            

data/ohlcv_2023-09.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-09.parquet


data/ohlcv_2023-08.parquet: reconstructing file:   0%|          |  0.00B /  348MB            

data/ohlcv_2023-08.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-08.parquet


data/ohlcv_2023-07.parquet: reconstructing file:   0%|          |  0.00B /  303MB            

data/ohlcv_2023-07.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-07.parquet


data/ohlcv_2023-06.parquet: reconstructing file:   0%|          |  0.00B /  332MB            

data/ohlcv_2023-06.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-06.parquet


data/ohlcv_2023-05.parquet: reconstructing file:   0%|          |  0.00B /  345MB            

data/ohlcv_2023-05.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-05.parquet


data/ohlcv_2023-04.parquet: reconstructing file:   0%|          |  0.00B /  288MB            

data/ohlcv_2023-04.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-04.parquet


data/ohlcv_2023-03.parquet: reconstructing file:   0%|          |  0.00B /  352MB            

data/ohlcv_2023-03.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-03.parquet


data/ohlcv_2023-02.parquet: reconstructing file:   0%|          |  0.00B /  313MB            

data/ohlcv_2023-02.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-02.parquet


data/ohlcv_2023-01.parquet: reconstructing file:   0%|          |  0.00B /  329MB            

data/ohlcv_2023-01.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2023-01.parquet


data/ohlcv_2022-12.parquet: reconstructing file:   0%|          |  0.00B /  346MB            

data/ohlcv_2022-12.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2022-12.parquet


data/ohlcv_2022-11.parquet: reconstructing file:   0%|          |  0.00B /  350MB            

data/ohlcv_2022-11.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2022-11.parquet


data/ohlcv_2022-10.parquet: reconstructing file:   0%|          |  0.00B /  360MB            

data/ohlcv_2022-10.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2022-10.parquet


data/ohlcv_2022-09.parquet: reconstructing file:   0%|          |  0.00B /  354MB            

data/ohlcv_2022-09.parquet: downloading bytes:           |  0.00B            

/root/.cache/huggingface/hub/datasets--ggaddam--OHLCV-1m/snapshots/ef1551b11d8ee3e35c7521cf36deee155b43e77d/data/ohlcv_2022-09.parquet


**1.3. Displaying the working dataset into a dataframe**

In [15]:
import pandas as pd

df = pd.read_parquet(file_path)
print("Shape:", df.shape)
display(df.head())

Shape: (27792952, 7)


,timestamp,open,high,low,close,volume,ticker
0,2022-09-01 13:31:00+00:00,10.02,10.02,10.02,10.02,125.0,HMACU
1,2022-09-07 16:07:00+00:00,10.02,10.02,10.02,10.02,7510.0,HMACU
2,2022-09-09 13:37:00+00:00,10.03,10.03,10.03,10.03,9000.0,HMACU
3,2022-09-09 13:39:00+00:00,10.03,10.03,10.03,10.03,6000.0,HMACU
4,2022-09-09 19:19:00+00:00,10.03,10.03,10.03,10.03,800.0,HMACU


**Observations:**

- There's approximately 27 millions lines in the working dataset with 7 columns
- The dataset represent a 1 min OHLCV (Opening, High, Low, Closing and Volume)

## **2. Descriptive Statistics**

## **3. Visualization Strategies**

## **4. Outlier Detection**